### ISSUE TITLE
Add summary-name rule

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4510]
Axe-core should be able to fail this. i think we need a new rule for it:

```html
<details>
  <summary></summary>
  <p>Hello world</p>
</details>
```

**Comment by padmavemulapati** [comment]
Verified with the latest dev branch code, seeing `summary-name` rule fail for the code snippet 
```
<details>
  <summary></summary>
  <p>Hello world</p>
</details>
```

<img width="1671" alt="image" src="https://github.com/dequelabs/axe-core/assets/47409579/18bd96d7-043e-48cc-b61e-6fac97ffe9fc">

As this is the new rule in axe-core4.10.0, we cannt see this fail with the earlier versions:

![image](https://github.com/dequelabs/axe-core/assets/47409579/baf3e174-53d9-4fb1-9745-f2ca89361530)

## Environment:
| **Label**                        | **Value** |
| -------------------------- | ------ |
| **Product**               | _axe-core_ |
|**Version** | _4.10.0(Dev Branch Code base_ |
| **OS-Details**     | _MAC- 14.4.1 _ |
|**BrowserDetails**|_Chrome Version 126.0.4_|
